code/modules/mob/living/simple_animal/friendly/cat.dm

```diff
@@ -204,6 +204,30 @@
 		if("harm")
 			wuv(-1, M)
 
+/mob/living/simple_animal/pet/cat/attack_alien(mob/living/carbon/alien/humanoid/M)	//what the xeno was after all along in Alien
+	if(M.a_intent == "help" && M.caste == "h")
+		if (icon_state == "cat" || icon_state == "cat_rest" || icon_state == "cat_sit")
+			visible_message("<span class='name'>[M]</span> picks up and hugs <span class='name'>[src]</span>.")
+			src.Move(M)
+
+			M.sprite_changed_for_emote = TRUE
+			var/original_icon = M.icon
+			var/original_icon_state = M.icon_state
+
+			M.icon = 'icons/mob/alien_emotes.dmi'
+			M.icon_state = "cat_cuddle"
+
+			spawn(M.emote_length)

```

Comment:
`spawn()` is considered obsolete in favor of `addtimer`. In this case, you'd separate the code after this `spawn` into its own proc and have a timer to that. You can view existing examples for syntax.